### ISSUE TITLE
this.transitionTo('error') does nothing and reports nothing

### ISSUE THREAD

**Issue by adam-knights** [emberjs/ember.js#12624]
I want to transition to the error route myself. Within routes/application.js I try

actions: {
  customErrorAction: function() {
    this.transitionTo('error');
  }
}

No transition happens, no errors appear in console etc. Almost like it treats it as a no op?

Example twiddle here https://ember-twiddle.com/f2587863883b1019d245.

**Comment by adam-knights** [comment]
If you do something like `this.transitionTo('error', 'TTTT');` it works. Its whether this.transitionTo('error') should also work, or not work and provide an error or warning, or is behaving as expected?

**Comment by pixelhandler** [comment]
@adam-knights I think `transitionTo('error', errObj)` with some context, e.g. the error object is the intended behavior.

**Comment by adam-knights** [comment]
@pixelhandler agreed that's what is wanted, but do you think rather than doing nothing with the single argument it should show a warning or error?

**Comment by pixelhandler** [comment]
@adam-knights well, the api docs do say the `model` argument to `transitionTo` is optional, http://emberjs.com/api/classes/Ember.Route.html#method_transitionTo

But, in practice… I've found that I **either** use my own routes as error handing, e.g. 'not-found', 'server-error' etc. **or** use the error sub-state _default_ behavior (no call to transition to any 'error' sub-state in my code).

There seems to be some 'special' handing of the error substates that is not obvious via guides or api docs. Personally I don't think it is a good practice to mix catching an error and transition to the default error sub-states. This is a portion of the ember codebase that could use some more documentation (guides) and/or improvements to the implementation e.g. using `transitionTo('error')` in your own code.

**Comment by stefanpenner** [comment]
cc @machty

**Comment by gnapse** [comment]
Apparently I'm at the exact same point described by @pixelhandler above. I'm tempted to create my own routes for error handling, because I'm having a hard time programmatically triggering the error substates.

**Comment by nathanhammond** [comment]
@gnapse Can you describe exactly what it is you're trying to do? Not how (`transitionTo`) but instead what the user flow is?

**Comment by gnapse** [comment]
Sure @nathanhammond, here's the deal:

I have a login page, and when the user submits the form, I catch the submit and I perform an ajax request for signing in. The only normal failure I expect from that ajax request is 401 (unauthorized). If that ajax request ends up in 500 or 404 or any other error condition, I want to switch to an error page. Initially I wanted to achieve showing the same error page that shows up when the route's model hook fails (`app/templates/application-error`). But I couldn't achieve loading that programmatically. I can only get there when a route model hook fails.

**Comment by nathanhammond** [comment]
Okay, the thing to note here is that [_error routes actually have dynamic segments_](https://github.com/emberjs/ember.js/blob/master/packages/ember-routing/lib/system/dsl.js?source=c#L21).

This means that in order to achieve your goal of reusing the error page you must:
1. Use [`intermediateTransitionTo`](http://emberjs.com/api/classes/Ember.Route.html#method_intermediateTransitionTo) so that you don't get a new URL.
2. Pass the second `models` argument: `this.intermediateTransitionTo('application_error', true);`
3. Correctly address the error route name, they're always `${name}_error`.

Should be closed as not a bug. @pixelhandler possibly worth improving documentation for this, but seems more power-user-ish.

**Comment by adam-knights** [comment]
@nathanhammond should this not warn or error for my original twiddle/case?

**Comment by nathanhammond** [comment]
@adam-knights [This behaves identically to other routes with dynamic segments linked in this manner.](https://ember-twiddle.com/282a7a5d83d05e2ed06d5ee7cc624c83?openFiles=router.js%2C)

I'm not opposed to a new error message. Thoughts, larger audience of Internet people?

**Comment by adam-knights** [comment]
@nathanhammond Interesting. Would welcome others thoughts on the error message. I think this particular piece of doc is what adds to my confusion https://guides.emberjs.com/v2.8.0/routing/loading-and-error-substates/#toc_the-code-error-code-event - it suggests you can just navigate to the error route in this way.

**Comment by nathanhammond** [comment]
@adam-knights et al. note that `this.transitionTo('error-page');` will work if `error-page` is a static route defined in the application. It is not one of the special Ember routes. That being said, the docs there likely confuse the issue.

**Comment by rwjblue** [comment]
Yes, we should absolutely error better in this scenario.  As a matter of fact (after some router spelunking) I found [exactly the error we want](https://github.com/tildeio/router.js/blob/f3ff6564cb817b2cdb620ab430964f19e67c4919/lib/router/transition-intent/named-transition-intent.js#L143) as a comment 😈 .

I don't fully understand what the preceding comment is referring to:

> Ideally we should throw this error to provide maximal
> information to the user that not enough context objects
> were provided, but this proves too cumbersome in Ember
> in cases where inner template helpers are evaluated
> before parent helpers un-render, in which cases this
> error somewhat prematurely fires.

We can try uncommenting that error and see what happens 😸 .

**Comment by nathanhammond** [comment]
@rwjblue That line has been in existence commented as of [October 20, 2013](https://github.com/tildeio/router.js/commit/5f59a112f5686f08a7a094957c1e8b10e414cca0#diff-b9ec6e1e02ece39d3654767d2c3e5ddeR129) I'd say that we've possibly fixed the corresponding issue in the intervening three years. Interestingly though, that error message has never been uncommented. It first appears in the repo in that commit.

**Comment by rickieshi** [comment]
Just bumped into this error today with Ember 2.18, and dug through code before I found this thread. Can we uncomment [this line](https://github.com/tildeio/router.js/blob/master/lib/router/transition-intent/named-transition-intent.js#L209) so that proper error can be thrown?

**Comment by pixelhandler** [comment]
@adam-knights @gnapse @locks @nathanhammond @rickieshi @rwjblue @stefanpenner is this still an issue, perhaps we should close or create a new reproduction of this, what do you think?
